### ISSUE TITLE
support downloading modpacks from the new ftb app

### ISSUE THREAD

**Issue by skyrina** [PrismLauncher/PrismLauncher#1540]
### Role

i play ftb modpacks

### Suggestion

i want prism launcher to support downloading ftb modpacks

### Benefit

overwolf bad

### This suggestion is unique

- [X] I have searched the issue tracker and did not find an issue describing my suggestion, especially not one that has been rejected.

### You may use the editor below to elaborate further.

here's a really quick proof of concept written in js for reference
this doesn't call any private apis so there's no access token needed, hopefully someone can implement this into the launcher itself, have fun

```js
import { mkdir, writeFile } from 'node:fs/promises';
import { join } from 'node:path';

const funny = await fetch('https://api.modpacks.ch/public/modpack/103/6561/');
const data = await funny.json();
const len = data.files.length;

const root = './modpack/';
await mkdir(root, { recursive: true });
let i = 0;

console.log('downloading', data.files.length, 'files');

for (let file of data.files) {
	try {
		await mkdir(join(root, file.path), { recursive: true });
		let data;
		if (file.url) data = Buffer.from(await (await fetch(file.url)).arrayBuffer());
		if (file.curseforge) {
			const id = file.curseforge.file.toString();
			data = Buffer.from(await ((await fetch(encodeURI(`https://mediafilez.forgecdn.net/files/${id.substring(0, 4).replace(/^0+/gi, '')}/${id.substring(4).replace(/^0+/gi, '')}/${file.name}`).replace(/\+/gi, '%2B'))).arrayBuffer())); // cool one liner
		}
		if (data == undefined) throw "no data found";
		await writeFile(join(root, file.path, file.name), data);
		console.log('downloaded', file.name);
		console.log(len - ++i, 'files left');
	} catch (e) {
		console.error(e);
		console.log(file);
	}
}
```

**Comment by Trial97** [comment]
Hi @skyrina,

The removal of the FTB app support was made at the request of the FTB Team.
For more information:https://prismlauncher.org/news/ftb-removal/ 
But fret not there will be an alternative: download the modpack using the new FTB App and then import it into prism.

Here is the issue that describes this: https://github.com/PrismLauncher/PrismLauncher/issues/235 and here is the PR that implements it: https://github.com/PrismLauncher/PrismLauncher/pull/1359

As this is yet to be released(planned for 8.0) you can try it using the nightly builds found here: https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop

**Comment by TheVekter** [comment]
> Hi @skyrina,
> 
> The removal of the FTB app support was made at the request of the FTB Team. For more information:https://prismlauncher.org/news/ftb-removal/ But fret not there will be an alternative: download the modpack using the new FTB App and then import it into prism.
> 
> Here is the issue that describes this: #235 and here is the PR that implements it: #1359
> 
> As this is yet to be released(planned for 8.0) you can try it using the nightly builds found here: https://nightly.link/PrismLauncher/PrismLauncher/workflows/trigger_builds/develop

This is not a feasible workaround, as it still requires you to install the FTB App to begin with, meaning Overwolf gets a chance to install its spyware in the background. I shouldn't have to download a potentially dangerous program even temporarily to play a modpack.

**Comment by Trial97** [comment]
Hi @TheVekter,
If you say that the FTB App is `a potentially dangerous program` then how can the packs that they offer be safe?
Here is the short story:  they do not want other launchers to make financial gains over their work so they revoked the API usage. By downloading packs with their app, you see ads that generate money for them and they can create more amazing packs.
So until we do not have their approval to download the packs through Prism the current solution is the only solution.

**Comment by TheVekter** [comment]
> Hi @TheVekter, If you say that the FTB App is `a potentially dangerous program` then how can the packs that they offer be safe? Here is the short story: they do not want other launchers to make financial gains over their work so they revoked the API usage. By downloading packs with their app, you see ads that generate money for them and they can create more amazing packs. So until we do not have their approval to download the packs through Prism the current solution is the only solution.

The packs aren't the issue, it's Overwolf, the program they used to make FTB App. Prism wasn't making any money off their packs anyway, so it's not relevant.

**Comment by Trial97** [comment]
Well please take your time and read this article: https://prismlauncher.org/news/ftb-removal/
This should explain why we removed it.

**Comment by sTiKyt** [comment]
I guess we need a fork of prism now or something..

Dang i'm tired of this

First it MultiMC, then PolyMC, now Prism...

**Comment by TheVekter** [comment]
> I guess we need a fork of prism now or something..
> 
> Dang i'm tired of this
> 
> First it MultiMC, then PolyMC, now Prism...

FTB is putting out a version of the launcher that doesn't include Overwolf, so we can still use that to download packs without having to worry about putting adware/malware on our PC. Apparently the reason FTB can't just let third party launchers hook into their API is because CurseForge is making an exception for FTB's use of _their_ API, so third party launchers wouldn't be included in that.

tl;dr We don't, a fork isn't going to fix this issue.

**Comment by TheKodeToad** [comment]
We didn't have much choice. Since 8.0 you've been able to import packs which you have installed in the FTB App, unfortunately you just can't download them directly in the launcher.
